### ISSUE TITLE
Domain and recovery domain need an MDS

### ISSUE THREAD

**Issue by wimpers** [openvstorage/framework#791]
In case a domain and recovery domain is added, the mds amout should be 3 so there is a master and slave in the domain and a slave in the recovery domain (as initially implemented).

**Comment by khenderick** [comment]
Can already be configured in `/ovs/framework/storagedriver|mds_safety`. Todo: set the default to 3.

**Comment by khenderick** [comment]
Fixed by #859 packaged in openvstorage-2.7.2-rev.3911.91eb1ed by @wimpers

**Comment by kinvaris** [comment]
PASSED

Disk `test` is present on node WTjYf6j4St2hbfMy

```
e4ekzyFd8lbj7Ltj = 10.100.199.161, dd0132a1-e0b7-4f48-b152-62143c4bd429
WTjYf6j4St2hbfMy = 10.100.199.162, 4ed469ff-4e9c-456d-b452-87080978db32
YCEAH1GFCljOusXt = 10.100.199.163, 2c99ed0a-46b2-404b-8f4f-6366fc861f55
4NyN0mYOqJEc7dWO = 10.100.199.164, 4791e884-de5b-4da1-876f-0d71e0a5cd4c
```

No domains or recovery domains are registered yet:

```
root@ovs-node2:~# etcdctl get /ovs/framework/storagedriver
{"mds_maxload": 75, "mds_tlogs": 100, "mds_safety": 3}
```

```
In [1]: from ovs.dal.lists.vdisklist import VDiskList
In [2]: v = VDiskList.get_vdisks()[0]
In [8]: print len(v.mds_services)
3

In [9]: print v.mds_services[1]
{
    "is_master": false, 
    "mds_service_guid": "352e718b-01a7-4395-9613-52e0b7644dce", 
    "guid": "918b638f-90fb-4302-baf8-78da8fb8978c", 
    "vdisk_guid": "bf95b4f5-ec8c-4d90-9594-168a5370d283"
}
In [10]: print v.mds_services[2]
{
    "is_master": true, 
    "mds_service_guid": "ef313400-6178-4ccb-affd-b92fe9f57aaf", 
    "guid": "f0e62de4-586c-4baf-b622-3493e1330309", 
    "vdisk_guid": "bf95b4f5-ec8c-4d90-9594-168a5370d283"
}
In [11]: print v.mds_services[0]
{
    "is_master": false, 
    "mds_service_guid": "647e2431-75d3-4357-ad6a-2011e8cc3a5a", 
    "guid": "471544c8-25b3-4c85-a55c-76de5e600734", 
    "vdisk_guid": "bf95b4f5-ec8c-4d90-9594-168a5370d283"
}
```

Added 2 Domains, DC1 & DC2. 
- DC1: Node1, Node2 & Node3 (Recovery domain DC2)
- DC2: Node4

```
{
    "is_master": true, 
    "mds_service_guid": "647e2431-75d3-4357-ad6a-2011e8cc3a5a", 
    "guid": "471544c8-25b3-4c85-a55c-76de5e600734", 
    "vdisk_guid": "bf95b4f5-ec8c-4d90-9594-168a5370d283"
}
{
    "is_master": false, 
    "mds_service_guid": "352e718b-01a7-4395-9613-52e0b7644dce", 
    "guid": "918b638f-90fb-4302-baf8-78da8fb8978c", 
    "vdisk_guid": "bf95b4f5-ec8c-4d90-9594-168a5370d283"
}
{
    "is_master": false, 
    "mds_service_guid": "f4523bbe-fb78-4894-bf81-80b7d09bfe89", 
    "guid": "a8d9f627-95e9-4c05-bbc4-8a5f6199fb60", 
    "vdisk_guid": "bf95b4f5-ec8c-4d90-9594-168a5370d283"
}
```

As proven here, we can see that one slave has moved to datacenter 2 after creating the domains & recovery domains, 
while 1 slave has stayed behind in the datacenter where the disk is actually located:

```
In [26]: print MDSServiceList.get_by_storagedriver('2c99ed0a-46b2-404b-8f4f-6366fc861f55')
{
    "guid": "352e718b-01a7-4395-9613-52e0b7644dce", 
    "capacity": -1, 
    "number": 0, 
    "vpool_guid": "9fcad835-3ffb-4656-9b57-3ef6073d20b4", 
    "service_guid": "24d31b5d-aae7-4d45-9c08-4f6ad4d80ce0"
}

In [27]: print MDSServiceList.get_by_storagedriver('4791e884-de5b-4da1-876f-0d71e0a5cd4c')
{
    "guid": "f4523bbe-fb78-4894-bf81-80b7d09bfe89", 
    "capacity": -1, 
    "number": 0, 
    "vpool_guid": "9fcad835-3ffb-4656-9b57-3ef6073d20b4", 
    "service_guid": "9c0f4bb6-5c08-4c66-8f70-95998548193d"
}

In [28]: print MDSServiceList.get_by_storagedriver('4ed469ff-4e9c-456d-b452-87080978db32')
{
    "guid": "647e2431-75d3-4357-ad6a-2011e8cc3a5a", 
    "capacity": -1, 
    "number": 0, 
    "vpool_guid": "9fcad835-3ffb-4656-9b57-3ef6073d20b4", 
    "service_guid": "0b1b1ffc-4df1-4fd6-87b8-42c1d961cb9e"
}

In [29]: print MDSServiceList.get_by_storagedriver('dd0132a1-e0b7-4f48-b152-62143c4bd429')
{
    "guid": "ef313400-6178-4ccb-affd-b92fe9f57aaf", 
    "capacity": -1, 
    "number": 0, 
    "vpool_guid": "9fcad835-3ffb-4656-9b57-3ef6073d20b4", 
    "service_guid": "d13213c0-6c10-447b-a87b-af18bb299537"
}
```
